ui/src/overview.tsx

```diff
@@ -117,9 +120,54 @@ function AddCriteriaButton(props: { group: string | GroupKind }) {
   );
 }
 
-function ParticipantsGroup(props: { group: Group }) {
+function ParticipantsGroup(props: { group: Group; index: number }) {
+  const dispatch = useAppDispatch();
+  const cohort = useCohort();
+  const [renameGroupDialog, showRenameGroup] = useTextInputDialog({
+    title: "Edit Group Name",
+    initialText:
+      props.group.name === ""
```

Comment:
Pull this entire expression out into a variable to ensure it stays consistent:
```
const groupName = props.group.name || "Group " + String(props.index + 1);
```

Reply:
Agree. yesterday, I also found there are twice this expression. But i do not have a good idea to pass the variable from front to back. that's good